Classes/Issues/Comments/Reactions/IssueCommentReactionCell.swift

```diff
@@ -127,27 +127,35 @@ UICollectionViewDelegateFlowLayout {
     }
 
     @objc private func onThumbsUp() {
-        delegate?.didAdd(cell: self, reaction: .thumbsUp)
+        toggle(reaction: .thumbsUp)
     }
 
     @objc private func onThumbsDown() {
-        delegate?.didAdd(cell: self, reaction: .thumbsDown)
+        toggle(reaction: .thumbsDown)
     }
 
     @objc private func onLaugh() {
-        delegate?.didAdd(cell: self, reaction: .laugh)
+        toggle(reaction: .laugh)
     }
 
     @objc private func onHooray() {
-        delegate?.didAdd(cell: self, reaction: .hooray)
+        toggle(reaction: .hooray)
     }
 
     @objc private func onConfused() {
-        delegate?.didAdd(cell: self, reaction: .confused)
+        toggle(reaction: .confused)
     }
 
     @objc private func onHeart() {
-        delegate?.didAdd(cell: self, reaction: .heart)
+        toggle(reaction: .heart)
+    }
+
+    func toggle(reaction: ReactionContent) {
+        if let didReact = reactions.first(where: { $0.content == reaction })?.viewerDidReact, didReact == true {
```

Comment:
I'm not crazy about this. We're putting biz logic into the view here.

Also, I think the menu should be an additive action only, and no-op if the reaction is already there (per #492).

I think just a `guard` checking that the reaction doesn't exist yet would be good here:

https://github.com/rnystrom/GitHawk/blob/master/Classes/Issues/Comments/IssueCommentSectionController.swift#L256

Reply:
Agree on the biz logic. The no-op would be a deviation from how GitHub does it now, though. That ok?

Reply:
Oooh didn't realize that's what GH does. Hmm. Maybe let's stick w/ how GH does it, just move the biz logic into the controller?